pages/login/index.js

```diff
@@ -0,0 +1,234 @@
+
+
+  const signupbutton = createNode("div", {"class": "button ctaPrimary", "style": "margin-top: 10px", "id": "sign-in-button"}, "Create account")
+  
+  formDiv.appendChild(nameInput)
+  formDiv.appendChild(emailInput)
+  formDiv.appendChild(passwordInput)
+  formDiv.appendChild(signupbutton)
+
+  signupbutton.addEventListener('click', () => signup())
+}
+
+function updateLogin () {
+  //clear current form
+  if (_is_signin_screen) {
+    // update to sign up screen
+    add_signup_form()
+    document.querySelector('#sign-in-type-text').textContent = 'Have an account?'
+    document.querySelector('#sign-in-type-button').textContent = 'Sign in'
+  } else {
+    add_signin_form()
+    document.querySelector('#sign-in-type-text').textContent = 'No account?'
+    document.querySelector('#sign-in-type-button').textContent = 'Sign up'
+  }
+
+  _is_signin_screen = !_is_signin_screen
+}
+
+// start listening for extension specific information
+window.addEventListener("message", receiveMessage)
+function receiveMessage (event) {
+    const message = event.data
+    logInfo(`${_LOG_SCOPE} Receiving message of type ${message['type']}`)
+    switch (message['type']) {
+      case MESSAGE_TYPES.APP_TO_EXTERNAL_SET_EXTENSION_INFO:
+        _extensionId = message['extensionId']
+        _forceServicesHostname = message['forceServicesHostname']
+        break
+      case MESSAGE_TYPES.APP_TO_EXTERNAL_CHECK_IS_LOADED:
+        window.postMessage({'type': MESSAGE_TYPES.EXTERNAL_TO_APP_IS_LOADED})
+        break
+      default:
+        logInfo(`${_LOG_SCOPE} Skipping message of type ${message['type']}`)
+        break
+    }
+}
+window.postMessage({'type': MESSAGE_TYPES.EXTERNAL_TO_APP_IS_LOADED})
+
+function addBanner(bannerText, isError) {
+  const bannerBox = document.querySelector('#banner-box')
+  bannerBox.textContent = bannerText
+  bannerBox.style.display = 'flex';
+  bannerBox.style.backgroundColor = isError ? 'red' : 'green'
+  setTimeout(() => bannerBox.style.display = 'none', 2000)
+}
+
+
+// TODO - this is a hacky fix so we don't have to update the extension 
+// we should be passing a url in the parameters upon redirect to the login page so we know
+// if you came from the extension widget or upon installing the extension 
+function onSuccessRedirectBasedOfPageWidth() {
+  const pageWidth = getWidth()
+  if (pageWidth < 350) { 
+    // leave some margin (we officially set the coaching width to 300 px)
+    window.location.href = '/pages/live/index.html'
+  } else {
+    window.location.href = '/pages/start/index.html'
+  }
+}
+
+function setApiKey(apiKey) {
+  console.log(chrome)
+  chrome.runtime.sendMessage(_extensionId, {
+    'type': MESSAGE_TYPES.EXTERNAL_TO_BACKGROUND_SET_API_KEY, 
+    'apiKey': apiKey, 
+}, {}, (result) => {
+  // update the status. 
+  // note: it would be cleaner to await this as a promise instead of using a callback
+  //  but that seemed not to work for me
```

Comment:
i suspect it can be made work, let me know if you want to figure it out together. this isnt too hacky (or too central) that i care that strongly